examples/Clients/Chatter/Program.cs

```diff
@@ -39,10 +43,8 @@ static async Task<int> Main(string[] args)
             var name = args[0];
 
             // Server will only support Https on Windows and Linux
-            var credentials = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ChannelCredentials.Insecure : ClientResources.SslCredentials;
-
-            var channel = new Channel("localhost:50051", credentials);
-            var client = new Chatter.ChatterClient(channel);
+            var certificate = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? null : new X509Certificate2(Path.Combine(Resources.CertDir, "client.crt"));
+            var client = GrpcClientFactory.Create<Chatter.ChatterClient>(@"https://localhost:50051", certificate);
```

Comment:
In most gRPC languages, there's a concept of Channel (something heavyweight, roughly corresponding to a connection even though this parallel is not accurate due to channel laziness and e.g. loadbalancing) and a lightweight "stub" (the client class for given service).
We should consider exposing such concepts too if possible (Channel would be replaced by something that implements a CallInvoker).

Reply:
Figuring out this probably extends beyond the scope of this PR.